src/api/common/connection.js

```diff
@@ -0,0 +1,125 @@
+'use strict';
+const {EventEmitter} = require('events');
+const WebSocket = require('ws');
```

Comment:
I think there was something blocking us from using them, but I can't remember what it was. They might work now.